### PR TITLE
Support running compiler unit tests on older (read: <0.4.9) compilers

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
Tested with 0.2.0, 0.3.5-nightly.2016.6.14, 0.3.6, 0.4.5 and 0.4.20-nightly.2018.2.13.

Fails with 0.4.0 which seems to be quite a broken release (internal compiler errors for a lot of things).

Still fails with the matuselah 0.1.1 :)

**Comment by axic** [comment]
Works now with 0.1.1, 0.1.2, 0.1.3 and 0.1.4 too :)

**Comment by axic** [comment]
Most of the tests are disabled for <0.1.6 as they require multiple file support. Solution will be to write them as single source tests in the future.

File: translate.js [hunk 1]
```diff
@@ -8,7 +8,7 @@
 /// New style: 0.4.5+commit.b318366e.Emscripten.clang
 function versionToSemver (version) {
   // FIXME: parse more detail, but this is a good start
-  var parsed = version.match(/^([0-9]+\.[0-9]+\.[0-9]+)-([0-9a-f]{8})[\/\*].*$/);
```
Comment (chriseth) [inline translate.js]: I don't understand the description of the commit. This is just about not escaping the special symbols, which is not required here anyway.

Comment (axic) [reply translate.js]: That's it.

**Pull request merged**